src/CoreWCF.MSMQ/src/CoreWCF/Channels/NetMsmqBinding.cs

```diff
@@ -0,0 +1,149 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.ComponentModel;
+using System.Xml;
+using CoreWCF.Runtime;
+
+namespace CoreWCF.Channels
+{
+    public class NetMsmqBinding : MsmqBindingBase
+    {
+        // private BindingElements
+        private BinaryMessageEncodingBindingElement _encoding;
+        private NetMsmqSecurity _security;
+
+        public NetMsmqBinding()
+        {
+            Initialize();
+            _security = new NetMsmqSecurity();
+        }
+
+        public NetMsmqBinding(string configurationName)
+        {
+            Initialize();
+            _security = new NetMsmqSecurity();
+        }
+
+        public NetMsmqBinding(NetMsmqSecurityMode securityMode)
+        {
+            Initialize();
+            _security = new NetMsmqSecurity(securityMode);
+        }
+
+        private NetMsmqBinding(NetMsmqSecurity security)
+        {
+            Initialize();
+            Fx.Assert(security != null, "Invalid (null) NetMsmqSecurity value");
+            _security = security;
+        }
+
+        public XmlDictionaryReaderQuotas ReaderQuotas
+        {
+            get { return _encoding.ReaderQuotas; }
+            set
+            {
+                if (value == null)
+                    throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull(nameof(value));
+                value.CopyTo(_encoding.ReaderQuotas);
+            }
+        }
+
+        public NetMsmqSecurity Security
+        {
+            get { return _security; }
+            set { _security = value; }
+        }
+
+        public EnvelopeVersion EnvelopeVersion
+        {
+            get { return EnvelopeVersion.Soap12; }
+        }
+
+        // [DefaultValue(TransportDefaults.MaxBufferPoolSize)]
+        public long MaxBufferPoolSize
+        {
+            get { return _transport.MaxBufferPoolSize; }
+            set { _transport.MaxBufferPoolSize = value; }
+        }
+
+        internal int MaxPoolSize
+        {
+            get { return (_transport as MsmqTransportBindingElement).MaxPoolSize; }
+            set { (_transport as MsmqTransportBindingElement).MaxPoolSize = value; }
+        }
+
+        [DefaultValue(MsmqDefaults.UseActiveDirectory)]
+        public bool UseActiveDirectory
```

Comment:
does this work ? if not should we throw not implemented exception ?

Reply:
Yes, I added throw exception in MsmqTransportBindingElement